test_recommended_jobs.py

```diff
@@ -147,21 +142,38 @@ def test_find_job(self, mocked_request):
         mocked_request.return_value = Mock(text='{"soc":1115,"title":"Chief executives and senior officials"}',
                                            status_code=200)
 
-        actual_data = main.find_job(rev, job_num)
+        actual_data = recommend_jobs.find_job(rev, job_num)
 
         mocked_request.assert_called_with('https://api.lmiforall.org.uk/api/v1/soc/code/1', verify=False)
         self.assertEqual(expected_data, actual_data)
 
-    @mock.patch('requests.get')
-    def test_no_response_from_API(self, mocked_request):
-        expected_data = 'There seems to be a issue getting your job recommendations back to you, please try again later'
-        mocked_request.return_value = Mock(status_code=400)
-
-        with self.assertRaises(Exception) as context:
-            main.api_call('https://api.lmiforall.org.uk/api/v1/soc/search?q=' + 'plumber')
-
-        self.assertEqual(expected_data, str(context.exception))
-        mocked_request.assert_called_with('https://api.lmiforall.org.uk/api/v1/soc/search?q=plumber', verify=False)
+    @mock.patch('builtins.input', side_effect=['engineer', 'No', 'engineer'])
+    @mock.patch('recommend_jobs.get_soc_code')
+    @mock.patch('recommend_jobs.soc_to_onet')
+    @mock.patch('recommend_jobs.onet_skills')
+    @mock.patch('recommend_jobs.onet_interests')
+    @mock.patch('recommend_jobs.reverse_search')
+    @mock.patch('recommend_jobs.find_job')
+    def test_run(self, mocked_job, mocked_search, mocked_onet_interests,
+                 mocked_onet_skills, mocked_onet, mocked_soc, Mock):
+        expected_data = ['plumber']
+        mocked_job.return_value = ['plumber']
+        mocked_search.return_value = [1115]
+        mocked_onet_interests.return_value = [{'id': '1.B.1.a', 'name': 'Realistic', 'value': 1.33}]
+        mocked_onet_skills.return_value = [{'id': '2.A.1.a', 'name': 'Reading Comprehension', 'value': 3.0},
```

Comment:
Again, a lot of repetition.

Could you take a few skills/interests, or any of these jsons and put them as constant variables

Reply:
Done